hydra/core/override_parser/overrides_visitor.py

```diff
@@ -144,18 +149,18 @@ def visitDictValue(
     ) -> Dict[str, ParsedElementType]:
         ret = {}
         children = ctx.getChildren()
-        assert self.is_matching_terminal(next(children), "{")
+        assert self.is_matching_terminal(next(children), OverrideLexer.BRACE_OPEN)
```

Comment:
this is leaking to error messages:
```
$ python ~/dev/hydra/examples/tutorials/basic/your_first_hydra_app/5_defaults/my_app.py key
Error parsing override 'key'
missing '=' at '<EOF>'
```

By using constants here, I think it would become:
```
$ python ~/dev/hydra/examples/tutorials/basic/your_first_hydra_app/5_defaults/my_app.py key
Error parsing override 'key'
missing EQUAL at '<EOF>'
```
Which is my opinion is more cryptic.
This is not a huge deal, but if possible I would rather keep the old error.

Reply:
You are right, though this is not related to this piece of code in the example you gave (that one is triggered by antlr, before even calling the visitor). It looks like it should be possible to customize antlr error messages by subclassing the parser. I'd rather do it in a subsequent PR though if that's ok (to avoid inflating this PR too much).

Reply:
Attempt in a followup PR is fine.